Fireworks-ai inference support 🔥 (#2677)
huggingface/blog: fireworks-ai.md

---
title: "Welcome Fireworks.ai on the Hub 🎆"
thumbnail: /blog/assets/inference-providers/fireworks-ai.png
authors:

> - user: your_username

Author: TODO: need to add team members from https://huggingface.co/fireworks-ai

---

Following our recent announcement on [Inference Providers on the Hub](https://huggingface.co/blog/inference-providers), we're thrilled to share that **Fireworks.ai** is now a supported Inference Provider on HF Hub!

> Fireworks.ai delivers blazing-fast, serverless inference directly on your model pages—making it easier than ever to deploy and experiment with your favorite models.

Reviewer: ```suggestion
Fireworks.ai delivers blazing-fast serverless inference directly on model pages—making it easier than ever to deploy and experiment with your favorite models.
```

Light up your projects with Fireworks.ai today!

## How it works

### In the website UI

> 

Author: TODO: in this section keep the same flow as in the original blogpost

TODO(screenshot)

> See all models supported by Fireworks on HF here: https://huggingface.co/models?inference_provider=fireworks-ai

Reviewer: ```suggestion
Search for all models supported by Fireworks on HF [here](https://huggingface.co/models?inference_provider=fireworks-ai)
```

> #### from Python, using huggingface_hub

Reviewer: I think there are too many headings here. 

```suggestion
#### From your code

Use the `huggingface_hub` python library to call Fireworks.ai endpoints by defining the `provider` parameter.
```

> 

Reviewer 2: ```suggestion
Install `huggingface_hub` from source: 
```bash
pip install git+https://github.com/huggingface/huggingface_hub
```
```

```python
from huggingface_hub import InferenceClient

client = InferenceClient(
	provider="fireworks-ai",
	api_key="xxxxxxxxxxxxxxxxxxxxxxxx"
)

result = client.text_to_image(
	"A vibrant fireworks display",
	model=""
)
print(result)

```

> #### from JS using @huggingface/inference

Reviewer: ```suggestion
Also, from JS using the `@huggingface/inference` library and the `provider` parameter set to `"fireworks-ai"`.
```

```js
import { HfInference } from "@huggingface/inference";

const client = new HfInference("xxxxxxxxxxxxxxxxxxxxxxxx");

const result = await client.textToImage({
  prompt: "A vibrant fireworks display",
  provider: "fireworks-ai"
});

console.log(result);
```

> ### From HTTP calls

Reviewer: ```suggestion
Or just through any HTTP request.
```

> TODO

Reviewer: ```suggestion
```sh
curl 'https://router.huggingface.co/fireworks-ai/v1/chat/completions' \
-H 'Authorization: Bearer hf_xxxxxxxxxxxxxxxxxxxxxxxx' \
-H 'Content-Type: application/json' \
--data '{
    "model": "Qwen/Qwen2.5-72B-Instruct",
    "messages": [
		{
			"role": "user",
			"content": "What is the capital of France?"
		}
	],
    "max_tokens": 500,
    "stream": false
}'
```
```

## Billing

For direct requests, i.e. when you use a Fireworks key, you are billed directly on your Fireworks account.

> For routed requests, i.e. when you authenticate via the hub, you'll only pay the standard Fireworks API rates. There's no additional markup from us, we just pass through the provider costs directly. (In the future, we may establish revenue-sharing agreements with our provider partners.)

Reviewer: Should we keep the PRO subs CTA from the main post?

```suggestion
For routed requests, i.e. when you authenticate via the hub, you'll only pay the standard Fireworks API rates. There's no additional markup from us, we just pass through the provider costs directly. (In the future, we may establish revenue-sharing agreements with our provider partners.)

Important Note ‼️ PRO users get $2 worth of Inference credits every month. You can use them across providers. 🔥

Subscribe to the [Hugging Face PRO plan](https://hf.co/subscribe/pro) to get access to Inference credits, ZeroGPU, Spaces Dev Mode, 20x higher limits, and more.
```

---
title: "Welcome Fireworks.ai on the Hub 🎆"
thumbnail: /blog/assets/inference-providers/fireworks-ai.png
authors:
- user: your_username
---

Following our recent announcement on [Inference Providers on the Hub](https://huggingface.co/blog/inference-providers), we're thrilled to share that **Fireworks.ai** is now a supported Inference Provider on HF Hub!

> [Fireworks.ai](https://fireworks.ai) delivers blazing-fast, serverless inference directly on your model pages—making it easier than ever to deploy and experiment with your favorite models.

Reviewer 3: ```suggestion
[Fireworks.ai](https://fireworks.ai) delivers blazing-fast, serverless inference directly on model pages in the Hub—making it easier than ever to deploy and experiment with your favorite models.
```
Not very happy with my suggestion, but `your model pages` sounded a bit confusing to me.

Reviewer 4: or simply?

```suggestion
[Fireworks.ai](https://fireworks.ai) delivers blazing-fast, serverless inference directly on model pages—making it easier than ever to deploy and experiment with your favorite models.
```

Reviewer 5: I like @Wauplin's suggestion - less wordy!

Among others, starting now, you can run serverless inference to the following models via Fireworks.ai:

- deepseek-ai/DeepSeek-R1
- deepseek-ai/DeepSeek-V3

> - mistralai/Mistral-Small-24B-Instruct-2501

Reviewer 4: maybe also mention a VLM from https://huggingface.co/api/partners/fireworks-ai/models as well ?
```suggestion
- mistralai/Mistral-Small-24B-Instruct-2501
- meta-llama/Llama-3.2-90B-Vision-Instruct
```

Light up your projects with Fireworks.ai today!

## How it works

### In the website UI

TODO(screenshot)

See all models supported by Fireworks on HF here: https://huggingface.co/models?inference_provider=fireworks-ai

#### from Python, using huggingface_hub

> ```python
> from huggingface_hub import InferenceClient
> 
> client = InferenceClient(
> 	provider="fireworks-ai",
> 	api_key="xxxxxxxxxxxxxxxxxxxxxxxx"
> )
> 
> result = client.text_to_image(
> 	"A vibrant fireworks display",
> 	model=""
> )
> print(result)
> 
> ```

Reviewer 2: fireworks-ai supports only conversational task

```suggestion
```python
from huggingface_hub import InferenceClient

client = InferenceClient(provider="fireworks-ai", api_key="xxxxxxxxxxxxxxxxxxxxxxxx")

output = client.chat_completion(
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "What is deep learning?"},
    ],
    model="meta-llama/Llama-3.3-70B-Instruct",
    max_tokens=20,
)
print(output.choices[0].message.content)
```
```

Reviewer 5: Fixed it in the latest commit - sorry was in the process of updating it lmao

[Fireworks.ai](https://fireworks.ai) delivers blazing-fast, serverless inference directly on your model pages—making it easier than ever to deploy and experiment with your favorite models.

Among others, starting now, you can run serverless inference to the following models via Fireworks.ai:

- [deepseek-ai/DeepSeek-R1](https://huggingface.co/deepseek-ai/DeepSeek-R1)
- [deepseek-ai/DeepSeek-V3](https://huggingface.co/deepseek-ai/DeepSeek-V3)
- [mistralai/Mistral-Small-24B-Instruct-2501](https://huggingface.co/mistralai/Mistral-Small-24B-Instruct-2501)
- [Qwen/Qwen2.5-Coder-32B-Instruct](https://huggingface.co/Qwen/Qwen2.5-Coder-32B-Instruct)
- [meta-llama/Llama-3.2-90B-Vision-Instruct](https://huggingface.co/meta-llama/Llama-3.2-90B-Vision-Instruct)

and many more, you can find the full list [here](https://huggingface.co/models?inference_provider=fireworks-ai).

Light up your projects with Fireworks.ai today!

## How it works

### In the website UI

TODO(screenshot)

See all models supported by Fireworks on HF here: https://huggingface.co/models?inference_provider=fireworks-ai

> #### from Python, using huggingface_hub
> Install `huggingface_hub` from source: 
> ```bash
> pip install git+https://github.com/huggingface/huggingface_hub
> ```python
> from huggingface_hub import InferenceClient
> 

Reviewer 4: ```suggestion
#### from Python, using huggingface_hub

Install `huggingface_hub` from source: 

```bash
pip install git+https://github.com/huggingface/huggingface_hub
```

```python
from huggingface_hub import InferenceClient

```
